### PR TITLE
chore: add apisixconsumer data structures

### PR DESCRIPTION
**Pull request by tokers**
Please answer these questions before submitting a pull request

This Pull Request adds the basic CRD and go struct data structures.

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues

___
### Bugfix
- Description

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

___
### Backport patches
- Why need to backport?

- Source branch

- Related commits and pull requests

- Target branch

### CODE REVIEW

File: pkg/kube/apisix/apis/config/v2alpha1/types.go [hunk 1]
```diff
@@ -309,3 +310,56 @@ type ApisixClusterConfigList struct {
...
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+
+// ApisixConsumer is the Schema for the ApisixConsumer resource.
+// An ApisixConsumer is used to identify a consumer.
+type ApisixConsumer struct {
+	metav1.TypeMeta   `json:",inline" yaml:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+	Spec              ApisixConsumerSpec `json:"spec,omitempty" yaml:"spec,omitempty"`
+}
```
Comment (gxthrj) [inline pkg/kube/apisix/apis/config/v2alpha1/types.go:324]: Add `Status` for ApisixConsumer , e.g. [ApisixRoute](https://github.com/apache/apisix-ingress-controller/blob/master/pkg/kube/apisix/apis/config/v2alpha1/types.go#L68)

Comment (tokers) [reply pkg/kube/apisix/apis/config/v2alpha1/types.go:324]: We can add it in the future.

Comment (gxthrj) [reply pkg/kube/apisix/apis/config/v2alpha1/types.go:324]: But you have added the `// +kubebuilder:subresource:status` for the ApisixConsumer.

Comment (tokers) [reply pkg/kube/apisix/apis/config/v2alpha1/types.go:324]: Got it, will add it as soon as possible.

Comment (tokers) [reply pkg/kube/apisix/apis/config/v2alpha1/types.go:324]: I found I already add it.

**Comment by tokers** [comment]
@gxthrj Please look it again.

File: pkg/kube/apisix/apis/config/v2alpha1/types.go (hunk 1 above)
Comment (gxthrj) [reply pkg/kube/apisix/apis/config/v2alpha1/types.go:324]: I mean you have added `// +kubebuilder:subresource:status`, but no `Status` field in ApisixConsumer struct.

Comment (tokers) [reply pkg/kube/apisix/apis/config/v2alpha1/types.go:324]: Added.

**Review by gxthrj — APPROVED** [review]

**Review by tao12345666333 — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**